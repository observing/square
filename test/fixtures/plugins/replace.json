{
  "globals": {
    "content": "var test = '{{replace-me}}'",
    "metrics": false,
    "distribution": "min",
    "engines": {
      "engine": "closure"
    }
  },
  "error": {
    "replace": {
      "replace-me": "I will replaced all the times!"
    }
  },
  "partial": {
    "replace": {
      "dev": {
        "replace-me": "I will replace only in development results!"
      }
    }
  },
  "full": {
    "replace": {
      "dev": {
        "replace-me": "I will replace only in development results!"
      },
      "min": {
        "replace-me": "I will replace only in minified results!"
      }
    }
  },
  "regex": {
    "replace": {
      "min": {
        "replace-me": {
          "regex": "/replace-me/",
          "value": "I was found by a RegExp"
        }
      }
    }
  },
  "regexfail": {
    "replace": {
      "min": {
        "replace-me": {
          "regex": "(fail to close the parentheses",
          "value": "I was found by a RegExp"
        }
      }
    }
  },
  "missing": {
    "replace": {
      "min": {
        "replace-me": {
          "regex": "lacking replace value"
        }
      }
    }
  }
}
