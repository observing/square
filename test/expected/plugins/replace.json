{
  "full": "var test = 'I will replace only in minified results!'",
  "regexp": "var test = 'I was found by a RegExp'",
  "double": "var test = 'I will replace only in minified results!'\nvar test = 'I will replace only in minified results!'",
  "regexfail": "Error: Invalid regular expression: SyntaxError: Invalid regular expression: /{{(fail to close the parentheses}}/: Unterminated group",
  "missing": "Error: Provide a proper regex and value if your using replace with regular expressions. Value for key `replace-me` is not replaced."
}
